middleware/ratelimiter/store.go

```diff
@@ -0,0 +1,86 @@
+package ratelimiter
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+type limiter struct {
+	requestQuota  int
+	counter       int
+	quotaDuration time.Duration
+	resetsAt      time.Time
+	mx            *sync.RWMutex
+}
+
+func newLimiter(requestQuota int, quotaDuration time.Duration) *limiter {
+	return &limiter{
+		mx:            new(sync.RWMutex),
+		requestQuota:  requestQuota,
+		quotaDuration: quotaDuration,
+		counter:       0,
+		resetsAt:      time.Now().Add(quotaDuration),
+	}
+}
+
+func (l *limiter) reset() {
+	l.mx.Lock()
+	defer l.mx.Unlock()
+	l.resetsAt = time.Now().Add(l.quotaDuration)
+	l.counter = 0
+}
+
+func (l *limiter) increment() {
+	l.mx.Lock()
+	defer l.mx.Unlock()
+	l.counter++
+}
+
+func (l *limiter) hasExpired() bool {
+	l.mx.RLock()
+	defer l.mx.RUnlock()
+	return time.Now().After(l.resetsAt)
+}
+
+func (l *limiter) hasExceededRequestQuota() bool {
+	l.mx.RLock()
+	defer l.mx.RUnlock()
+	return l.counter >= l.requestQuota
+}
+
+func (l *limiter) getRemainingRequestQuota() int {
+	l.mx.RLock()
+	defer l.mx.RUnlock()
+	return l.requestQuota - l.counter
+}
+
+func (l *limiter) getSecondsToQuotaReset() float64 {
+	l.mx.RLock()
+	defer l.mx.RUnlock()
+	return -1 * math.Round(time.Since(l.resetsAt).Seconds())
+}
+
+type limiterStore struct {
+	mx    *sync.RWMutex
```

Comment:
```suggestion
	mx    sync.RWMutex
```
Same thing, don't use a pointer for mutex.